Mathlib/FieldTheory/IntermediateField.lean

```diff
@@ -675,6 +675,43 @@ end RestrictScalars
 /-- This was formerly an instance called `lift2_alg`, but an instance above already provides it. -/
 example {F : IntermediateField K L} {E : IntermediateField F L} : Algebra K E := by infer_instance
 
+section ExtendScalars
+
+variable {F E E' : IntermediateField K L} (h : F ≤ E) (h' : F ≤ E') {x : L}
+
+/-- If `F ≤ E` are two intermediate fields of `L / K`, then `E` is also an intermediate field of
+`L / F`. It can be viewed as an inverse to `IntermediateField.restrictScalars`. -/
+def extendScalars : IntermediateField F L := E.toSubfield.toIntermediateField fun ⟨_, hf⟩ ↦ h hf
+
+@[simp]
+theorem coe_extendScalars : (extendScalars h : Set L) = (E : Set L) := rfl
+
+@[simp]
+theorem extendScalars_toSubfield : (extendScalars h).toSubfield = E.toSubfield :=
+  SetLike.coe_injective rfl
+
+@[simp]
+theorem mem_extendScalars : x ∈ extendScalars h ↔ x ∈ E := Iff.rfl
+
+@[simp]
+theorem extendScalars_restrictScalars : (extendScalars h).restrictScalars K = E := rfl
+
+theorem extendScalars_le_extendScalars_iff : extendScalars h ≤ extendScalars h' ↔ E ≤ E' := by rfl
+
+theorem extendScalars_le_iff (E' : IntermediateField F L) :
+    extendScalars h ≤ E' ↔ E ≤ E'.restrictScalars K := by rfl
+
+theorem le_extendScalars_iff (E' : IntermediateField F L) :
+    E' ≤ extendScalars h ↔ E'.restrictScalars K ≤ E := by rfl
```

Comment:
These could be `Iff.rfl`s.

Reply:
Yes, but `by rfl` has one character less than `Iff.rfl` 🤣